add comments of springboot 3.0 minimum JDK version (#3906)
open-telemetry/opentelemetry.io: content/en/docs/languages/java/instrumentation.md

{{% /alert %}}

> ## Prerequisites

Reviewer: as discussed below, can we move those prerequisites to within the sample app setup, and make it much more clearer that this is for that sample app

> ## Prerequisites {#prerequisites}

Reviewer 2: ```suggestion
## Prerequisites
```

> Ensure that you have the following installed locally:

Reviewer: Those prerequisites do only apply to the sample app, I would argue we move it there and call it out specifically.

I am wondering if we need a global prerequisites as well, to verify what minimum JDK version is needed overall?

Reviewer 3: ya, I share @jack-berg's https://github.com/open-telemetry/opentelemetry.io/pull/3877#discussion_r1469862533

> someone reading this may infer that we only support Java 17+

and https://github.com/open-telemetry/opentelemetry.io/pull/3877#discussion_r1469879885

> The JDK version compatibility is a pretty important point in the java ecosystem, so while the entire story may be documented elsewhere, I think any spot which mentions version requirement conflicting with java 8+ is prone to being misinterpreted

> - Java JDK 17+, due to the use of Spring Boot 3; [Java 8+ otherwise][java-vers]

Reviewer 2: You also need to bring over the link definition for `[java-vers]`.

You don't have to use the example app: if you want to instrument your own app or
library, follow the instructions here to adapt the process to your own code.

### Prerequisites

> Ensure that you have the following installed locally:

Reviewer: ```suggestion
For running the example application, ensure that you have the following installed locally:
```

> - Java JDK 17+, due to the use of Spring Boot 3; [Java 8+ otherwise][java-vers]

Reviewer: ```suggestion
- Java JDK 17+ (due to the use of Spring Boot 3, [OpenTelemetry Java works with Java 8+ otherwise][java-vers])
```

> - Java JDK 17+ (due to the use of Spring Boot 3, [OpenTelemetry Java works with Java 8+ otherwise][java-vers])

Reviewer 2: ```suggestion
- Java JDK 17+, due to the use of Spring Boot. OpenTelemetry Java only [requires Java 8+][java-vers] otherwise.
```

You don't have to use the example app: if you want to instrument your own app or
library, follow the instructions here to adapt the process to your own code.

### Prerequisites

For running the example app, ensure that you have the following installed
locally:

> - Java JDK 17+, due to the use of Spring Boot 3. OpenTelemetry Java only [requires

Reviewer 3: ```suggestion
- Java JDK 17+, due to the use of Spring Boot 3. OpenTelemetry Java itself only [requires
```

> - Java JDK 17+ (due to the use of Spring Boot 3, [OpenTelemetry Java works with
>   Java 8+ otherwise][java-vers])

Reviewer 2: ```suggestion
- Java JDK 17+, due to the use of Spring Boot. OpenTelemetry Java only 
  [requires Java 8+][java-vers] otherwise.
```

>   Java 8+][java-vers] otherwise.

Reviewer 2: > maybe also "Java JDK 17+" --> "Java 17+"?

That works for me.

As for the Spring Boot version: can we drop it so that we don't have to update the version down the line. The statement remains true without the version: Java 17+ is required _because_ of Spring Boot. WDYT @trask?

> - [Gradle](https://gradle.org/) or [Maven](https://maven.apache.org/)

Reviewer 4: The example app only has a gradle configuration. Its possible translate to the equivalent maven, but that doesn't exist right now.

Reviewer: good call out, thanks!

Reviewer: @pegasas please take a look at this open conversation

Reviewer: @pegasas can you take another look here? The simple solution is probably to drop the requirement for Maven, although the preferred solution would to have both
